### PR TITLE
Respect --new_group_enforce_role when implicitly creating groups

### PR DESCRIPTION
**Pull request by timcharper**
When creating apps such as "/dev/sleeper" in a non-existent group "/dev", Marathon will auto-create the necessary groups in order to facilitate the request. Previously, this behavior would ignore the enforceRole defaulting configuration, instead opting to always auto-create said groups with enforceRole: false (and consequently to not use, by default, the group-role for the role /dev/sleeper).

In this PR, we rectify the situation by enclosing the intent for new default groups in RootGroup. This state is threaded through from update to update.

Additionally, we update the defaulting logic for roles to use the configured newGroupRoleDefaultBehavior parameter.

JIRA issues: MARATHON-8706

### CODE REVIEW

File: src/main/scala/mesosphere/marathon/state/RootGroup.scala [hunk 1]
```diff
@@ -203,14 +209,14 @@ class RootGroup(
     *
     * Every transitive group gets the new version.
     *
-    * @param groupId the id of the group to be updated
+    * @param groupId      the id of the group to be updated
```
Comment (alenkacz) [inline src/main/scala/mesosphere/marathon/state/RootGroup.scala]: is this some new formatter feature that it tries to align the parameter descriptions? 🤔

File: src/main/scala/mesosphere/marathon/state/RootGroup.scala [hunk 2]
```diff
@@ -427,13 +440,14 @@ object RootGroup {
     pods: Map[AbsolutePathId, PodDefinition] = Group.defaultPods,
     groupsById: Map[AbsolutePathId, Group] = Group.defaultGroups,
     dependencies: Set[AbsolutePathId] = Group.defaultDependencies,
-    version: Timestamp = Group.defaultVersion): RootGroup = new RootGroup(apps, pods, groupsById, dependencies, version)
+    autoCreateStrategy: NewGroupStrategy = NewGroupStrategy.Fail,
```
Comment (alenkacz) [inline src/main/scala/mesosphere/marathon/state/RootGroup.scala]: slightly confused what is autocreate here, should it be newGroupStrategy?

Comment (alenkacz) [reply src/main/scala/mesosphere/marathon/state/RootGroup.scala]: also why is it fail by default? 🤔

Comment (ANeumann82) [reply src/main/scala/mesosphere/marathon/state/RootGroup.scala]: I renamed it to newGroupStrategy, as that name is used in other places as well.

Comment (timcharper) [reply src/main/scala/mesosphere/marathon/state/RootGroup.scala]: Thank you!

File: src/main/scala/mesosphere/marathon/state/RootGroup.scala (hunk 1 above)
Comment (timcharper) [reply src/main/scala/mesosphere/marathon/state/RootGroup.scala]: Maybe intellij did it

**Pull request merged**